backends/sycl/ceed-sycl-compile.sycl.cpp

```diff
@@ -96,7 +94,7 @@ static int CeedJitLoadModule_Sycl(Ceed ceed,
                                   ZE_MODULE_FORMAT_IL_SPIRV,
                                   il_binary.size(),
                                   il_binary.data(),
-                                  " -ze-opt-large-register-file",   // flags
+                                  " -g -ze-opt-large-register-file",
```

Comment:
We need to check if adding `-g` disables default optimization. We could add `-ze-opt-level 2` here just in case.